neuraxle/checkpoints/base_checkpoint_step.py

```diff
@@ -0,0 +1,57 @@
+from neuraxle.base import BaseStep
+
+
+class BaseCheckpointStep(BaseStep):
+    """
+    Base class for a checkpoint step that can persists the received data inputs, and expected_outputs
+    to eventually be able to load them using the checkpoint pipeline runner.
+    """
+
+    def fit(self, data_inputs, expected_outputs=None) -> 'BaseCheckpointStep':
+        """
+        Save checkpoint for data inputs and expected outputs so that it can
+        be loaded by the checkpoint pipeline runner on the next pipeline run
+        :param data_inputs: data inputs to save
+        :return: self
+        """
+        self.save_checkpoint(data_inputs, expected_outputs)
+
+        return self
+
+    def transform(self, data_inputs):
+        """
+        Save checkpoint for data inputs and expected outputs so that it can
+        be loaded by the checkpoint pipeline runner on the next pipeline run
+        :param data_inputs: data inputs to save
+        :return: data_inputs
+        """
+        self.save_checkpoint(data_inputs)
+
+        return data_inputs
+
+    def set_checkpoint_name(self, checkpoint_name):
+        """
+        Set checkpoint name.
+        :param checkpoint_name: str for checkpoint name
+        :return: tuple(data_inputs, expected_outputs)
+        """
+        raise NotImplementedError()
+
+    def load_checkpoint(self):
```

Comment:
Add `@abstractmethod` just above the def to force derived classes to override the method.